projects/components/src/time-range/time-range.component.test.ts

```diff
@@ -108,7 +108,7 @@ describe('Time range component', () => {
     expect(spectator.query('.refresh')).toExist();
   });
 
-  test('should not show refresh button when time range is fixed', () => {
+  test('should show refresh button when time range is fixed', () => {
```

Comment:
I think, given we are calling the other one `relative`, so here we are going as per convention in the code base, having `fixed` and `relative` time ranges, so going with `fixed` only.

Reply:
Also, jFYI, the relevant documentation for Angular is [here](https://the-guild.dev/graphql/apollo-angular/docs/caching/advanced-topics#resetting-the-store)

Reply:
Yes, but this only becomes clear after someone has seen the full code and already knows the two are called fixed and relative. For someone just looking at the UI, they could think fixed refers to the other one:

<img width="577" alt="Screenshot 2022-12-11 at 8 53 08 PM" src="https://user-images.githubusercontent.com/29686866/206912567-eae80f19-6a7f-49bb-9cb9-a346de4b15c3.png">

UI and code/tests should have 1:1 matching in names to avoid confusion whenever possible.

Reply:
I'd say I think differently about this. UI is more about putting labels to components, might have very different names per different instances, plus unit test cases are always linked to components. If you have a component called `fixed-time-range.component.ts`, your unit tests will sound too confusing if you write them for `custom`. It's your business logic that you use `fixed-time-range` for custom components, so, different perspective there.

Reply:
Also, the above image labels are wrong. The ones you have higlighted as `fixed-time-range` are actually `relative-time-range`. If you would go through the code once, you'd be able to see that the `fixed-time-range` is actually the one you see when you select custom. I'd suggest going through it once for clarity.

Reply:
> unit test cases are always linked to components

Shouldn't all tests be linked to UI instead? The [philosophy][1] of Testing Library is to avoid the mental model of tying your tests to components. I know [Ngneat][2] instead focuses on testing components directly so we're kinda handicapped by the tech stack but are we also following the philosophy of testing components instead of UI?

> the above image labels are wrong

That's the confusion that can arise. "Fixed" is an ambiguous term here. Is the top time range "fixed" because I can specify exact time, or are the bottom choices "fixed" because we can't change those. We can call both of them fixed depending on how we thought, but the word custom can only be assigned to the first one.

[1]: https://testing-library.com/docs/#what-you-should-avoid-with-testing-library
[2]: https://ngneat.github.io/spectator/

Reply:
The philosophy makes sense Testing library, I think that would make sense when it comes to non-enzyme type testing libraries, but here, most of the time, rather than rendering components and checking for views present on screen, tests are more about internals of the component, so, don't think that philosophy applies to us overall, in all aspects.

I don't think the confusion should arise if the code base is looked upon tbh. The problem with choosing custom as a keyword in the tests for me is that, tomorrow, you might call it `Specific` or `Explicit` time in some other use case, which might make the test case again confusing given it's a different label used. Also, there is an already existing `custom-time-range-selection.component.ts`, which houses datepickers and buttons within it. So that would again spawn a confusion about what the test it about and why might that be present here.

The test case, is applicable once the selection is done, and tests whether the `fixed-time-range` behaves as it should. Were it above selection part in the custom time, it would be more apt to name it in that fashion.